src/document/mod.rs

```diff
@@ -103,14 +104,11 @@ impl Document {
                             0
                         }
                     };
-                    let alt_block = if let Some(blocks) = code_blocks.get(&code_block.name) {
-                        blocks.get(index)
-                    } else {
-                        None
-                    };
+                    let alt_block = code_blocks.get(&(&code_block.name, &index));
+
                     output.push_str(
                         &printer
-                            .print_code_block_reverse(code_block, alt_block)
+                            .print_code_block_reverse(code_block, alt_block.copied())
```

Comment:
Not really sure how to best handle (or avoid) the `Option<&&RevCodeBlock>` here.